site/app/libraries/database/DatabaseQueries.php

```diff
@@ -6559,6 +6559,11 @@ public function getTimeJoinedQueue($user_id, $queue_code) {
         return $this->course_db->rows()[0]['max'];
     }
 
+    public function getStarType($user_id, $queue_code) {
+        $this->course_db->query("SELECT star_type FROM queue WHERE user_id = ? AND UPPER(TRIM(queue_code)) = UPPER(TRIM(?)) ORDER BY entry_id DESC", [$user_id, $queue_code]);
+        return $this->course_db->rows()[0]['star_type'];
```

Comment:
```suggestion
        return $this->course_db->row()['star_type'];
```

Reply:
Can this ever return more than one row?  You can use `LIMIT 1` in the query if you want to force it to only return one row.